plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/PinotQueryRunner.java

```diff
@@ -42,49 +44,84 @@ private PinotQueryRunner() {}
 
     public static final String PINOT_CATALOG = "pinot";
 
-    // TODO convert to builder
-    public static QueryRunner createPinotQueryRunner(
-            TestingKafka kafka,
-            TestingPinotCluster pinot,
-            Map<String, String> extraPinotProperties,
-            Iterable<TpchTable<?>> tables)
-            throws Exception
+    public static Builder builder()
     {
-        return createPinotQueryRunner(kafka, pinot, ImmutableMap.of(), extraPinotProperties, tables);
+        return new Builder();
     }
 
-    // TODO convert to builder
-    private static QueryRunner createPinotQueryRunner(
-            TestingKafka kafka,
-            TestingPinotCluster pinot,
-            Map<String, String> coordinatorProperties,
-            Map<String, String> extraPinotProperties,
-            Iterable<TpchTable<?>> tables)
-            throws Exception
+    public static class Builder
+            extends DistributedQueryRunner.Builder<Builder>
     {
-        QueryRunner queryRunner = DistributedQueryRunner.builder(createSession())
-                .setCoordinatorProperties(coordinatorProperties)
-                .build();
+        private TestingKafka kafka;
+        private TestingPinotCluster pinot;
```

Comment:
`pinot.controller-urls` can be defaulted when constructing the builder, so that the builder doesn't need a reference to the TestingPinotCluster (also makes it easier to configure pinot differently in the future)

Reply:
The both TestingKafka and TestingPinotCluster are used from `createTpchTables()` in `build()` method. How can we remove the reference?

Reply:
missed that